java/src/main/native/src/TableJni.cpp

```diff
@@ -775,6 +777,31 @@ jlongArray combine_join_results(JNIEnv *env, cudf::table &left_results,
   return combine_join_results(env, std::move(left_cols), std::move(right_cols));
 }
 
+struct scalar_construction_helper {
+  template <typename T,
+           typename ScalarType = cudf::scalar_type_t<T>,
+           typename std::enable_if_t<!(std::is_integral<T>() && !cudf::is_boolean<T>())>* = nullptr>
+  std::unique_ptr<cudf::scalar> operator()(int value, bool validify = true) {
+    CUDF_FAIL("Range window only supports non-bool integral type");
+  }
+
+  template <typename T,
+            typename ScalarType = cudf::scalar_type_t<T>,
+            typename std::enable_if_t<std::is_integral<T>() && !cudf::is_boolean<T>()>* = nullptr>
+  std::unique_ptr<cudf::scalar> operator()(int value, bool validify = true) {
```

Comment:
What happens if we want to store an int64_t value instead of an int, or what about an unsigned int32_t? The reason they were originally ints is because all we supported for time range based queries was days. That appears to no longer be the case.  @mythrocks for time stamp based range queries (not days) is the range value still only in days?

Reply:
Hi @revans2, Seems the condition of `std::is_integral<T>() && !cudf::is_boolean<T>()` can cover any integral type (int, long, unsigned int, unsigned long, char). The native PR @mythrocks made has supported integral type (boolean exclusive)

Reply:
That is exactly my point.  The JNI code is written assuming that the value is an `int` it cannot support any of the other types properly.

Reply:
Wow, I got your point, Thx for point it out, I will fix it.

Reply:
> @mythrocks for time stamp based range queries (not days) is the range value still only in days?

Sorry I missed your question, @revans2. For timestamp-based queries, the range values have to be specified as a duration-scalar that matches the timestamp's resolution.
Order By Type | Range Type
-------------------|---------------
TIMESTAMP_DAYS | DURATION_DAYS
TIMESTAMP_SECONDS | DURATION_SECONDS
TIMESTAMP_MILLISECONDS | DURATION_MILLISECONDS
TIMESTAMP_MICROSECONDS | DURATION_MICROSECONDS
TIMESTAMP_NANOSECONDS | DURATION_NANOSECONDS

Reply:
@revans2 @mythrocks  Done.